locals.tf

```diff
@@ -3,15 +3,27 @@ locals {
     enabled  = var.highly_available == true ? true : false
     location = "eu"
   }
+  default_insights_config = {
+    query_string_length     = 1024
+    record_application_tags = false
+    record_client_address   = false
+  }
   default_labels = {
     env = var.environment
   }
   default_tier = var.environment == "production" ? "db-custom-2-8192" : "db-f1-micro"
 
-  backup_config = defaults(var.backup_config, local.default_backup_config)
-  labels        = merge(local.default_labels, var.labels)
-  storage_size  = var.storage_autoresize == true ? null : var.storage_size
-  tier          = var.tier != null ? var.tier : local.default_tier
+  backup_config   = defaults(var.backup_config, local.default_backup_config)
+  insights_config = defaults(var.insights_config, local.default_insights_config)
```

Comment:
... and more locals ...

Reply:
And here the specified defaults fill in the blanks in the specified variables.
Since `default_insights_config` and `default_backup_config` are merely helper vars, I can replace the locals here with inline objects, but that's potentially less readable, given the length and number of options.